ngeeyonglim

ngeeyonglim opened the pull request:
# YONG chatbot

##Features
- List *feature*
- Find keyword feature
- Terminating keyword feature

##Possible **tasks** added
1. `Todo`
1. `Deadline`
1. `Event`

##Code for startup command
```java
public void execute() {
        String logo = "YONG";
        System.out.println("Hello I'm " + logo + "\n" +
                "What can I do for you? \n" +
                "-------------------------------------"
        );
    }
```

##Things you need to do 
- [x] ~~Like this pull request~~ :+1:
- [ ] Accept this pull request  

> This is my [IP](https://github.com/ngeeyonglim/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Rename, Greet, Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark as Done

Commit:
ToDo, Event, Deadline

Commit:
ToDo, Event, Deadline

Commit:
Automated Text UI Testing

Commit:
Handle Erros

Commit:
Delete

Commit:
Added unmark

Commit:
Added Java docs for all methods

Commit:
Add a save feature and refactor the Duke class name

Commit:
Add an example data file

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add an example data file
  Add a save feature and refactor the Duke class name

Commit:
Add function to parse dates and print in correct date format

Commit:
Modify action method to handle different cases

Commit:
Add support for Date and Time for specific tasks

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add support for Date and Time for specific tasks
  Modify action method to handle different cases
  Add function to parse dates and print in correct date format

Commit:
Make the code more OOP.

Extracts different classes and adds more encapsulation and abstraction to the code.

Commit:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Make the code more OOP.

Commit:
Add packages to separate different kinds of classes

Commit:
Merge branch 'A-Packages'

* A-Packages:
  Add packages to separate different kinds of classes

Commit:
Merge commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552'

* commit 'ed6d4d25bd4e34461919ffe785a6f10f4c087552':
  Bump gradle and lib version
  Add Gradle support

Commit:
Add junit tests for Parser and TaskList classes

Commit:
Merge branch 'A-JUnit'

* A-JUnit:
  Add junit tests for Parser and TaskList classes

Commit:
Merge 2 storage methods

Commit:
Change all imports to not include wildcards

Commit:
Change variable, methods and class names to coding standard

Commit:
Add line breaks between logical units

Commit:
Order import statements

Commit:
Change class attributes to be protected

Commit:
Edits javadocs

Commit:
Merge branch 'A-CodingStandard'

* A-CodingStandard:
  Change class attributes to be protected
  Order import statements
  Add line breaks between logical units
  Change variable, methods and class names to coding standard
  Change all imports to not include wildcards

Commit:
Merge branch 'A-JavaDoc'

* A-JavaDoc:
  Edits javadocs

Commit:
Add find command and related classes

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find command and related classes

# Conflicts:
#	src/main/java/duke/Parser.java

Commit:
Fix Storage class to create directory if needed

Commit:
Fix find method in Tasklist

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Fix find method in Tasklist

Commit:
Change find function in tasklist

georgetayqy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class Parser {
+
+    TaskList taskList;
```
Should this be marked as `private` instead?

georgetayqy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * Method to parse the string input of the chatbot and call the correct command.
```
Should this be phrased as `Parses the string input of the chatbot...`?

georgetayqy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+
+
+    /**
+     * Saves the current history array to the file
```
Just a small nit with the Javadocs:

```suggestion
     * Saves the current history array to the file.
```

georgetayqy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    /**
+     * Reads the file to populate the history array
```
Same small nit with the Javadocs:

```suggestion
     * Reads the file to populate the history array.
```

georgetayqy commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class UI {
+
+    static Scanner reader;
```
Should this have the `private` access modifier?

georgetayqy commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,54 @@
+    /**
+     * Prints out the error message.
+     * @param errorMessage Error message to be printed.
+     */
```
Another small nit with the Javadocs:

```suggestion
    /**
     * Prints out the error message.
     *
     * @param errorMessage Error message to be printed.
     */
```

georgetayqy commented on src/main/java/duke/Yong.java:
```diff
@@ -0,0 +1,43 @@
+        Yong.run();
+    }
+
+    public static void run() {
```
Should this method have Javadocs to explain what this method does?

georgetayqy commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public abstract class Command {
+
+    TaskList taskList;
```
Should this have the `private` access modifier? I notice that this issue can be seen in other parts of the codebase.

georgetayqy commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,41 @@
+
+    /**
+     * Abstract method to be implemented for all commands.
+     * To be run when the command is entered by the user.
```
Should this be phrased like `Runs when the command is entered by the user.`? I notice that this issue is also present in other parts of the codebase.

georgetayqy commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class DeadlineCommand extends Command {
+
+    String inp;
```
Should the variable name be more descriptive, such as `inputString`? I notice that this issue is also present in the other `*Command` classes.

georgetayqy commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,35 @@
+
+/**
+ * Represents the actions needed if the user inputs an Exit command.
+ * User input is "bye"
```
Should this line be removed?

georgetayqy commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,46 @@
+     * Constructor for the Find command.
+     *
+     * @param taskList TaskList object being used to maintain the list of tasks in the chatbot.
+     * @param inp      Line of input from the CLI
```
Should the extra whitespaces be removed?

georgetayqy commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,34 @@
+ */
+public class ListCommand extends Command {
+
+
```
Should these lines be removed?

georgetayqy commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Class to handle exceptions due to incorrect user input.
+ */
+public class DukeException extends RuntimeException {
```
Should this class be renamed to `DukeInputException` to better encapsulate the error that is thrown when the user enters in an incorrect input?

georgetayqy commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+     * Marks the specified task as being done.
+     *
+     * @param number Number of the task, starting from index 1.
+     * @return The task being marked as done.
```
Should the exception thrown be explictly mentioned here?

```suggestion
     * @param number Number of the task, starting from index 1.
     * @throws DukeException if the user inputs an index that is out of bounds
     * @return The task being marked as done.
```

georgetayqy commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+        this.taskList = taskList;
+    }
+
+    public void find(String lookupString) {
```
Should this method have Javadoc to explain what this method does?

georgetayqy commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+package duke.tasks;
+
+import java.time.LocalDateTime;
+/**
```
Should there be an empty line between the import statements and the Javadoc for the class?

georgetayqy commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime dueBy;
```
Should this be renamed to `dueByDate`?

georgetayqy commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,39 @@
+public class Event extends Task {
+
+    protected LocalDateTime from;
+    protected LocalDateTime to;
```
Should these be renamed to `fromDate` and `toDate`?

georgetayqy commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Test
+    public void parseTest() {
```
Should this be renamed to `testParse_validInputs_success`?

georgetayqy commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,58 @@
+            assertTrue(command instanceof EventCommand);
+
+            command = parser.parse("bye");
+            assertTrue(command instanceof ExitCommand);
```
Should these be placed in different test methods, to break up the test logic into separate standalone tests?

georgetayqy commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Test
+    public void invalidCommandTest() {
```
Should this be renamed to `testParse_invalidInputs_exceptionThrown`? I notice that this issue can be seen elsewhere in the codebase.

georgetayqy commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Marks the task as undone
+     */
+    public void unmarkAsDone() { this.isDone = false; }
```
Should this use the Egyptian-style brackets?

```suggestion
/**
 * Marks the task as undone
 */
public void unmarkAsDone() {
    this.isDone = false;
}
```

georgetayqy commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,96 @@
+        return date.format(DateTimeFormatter.ofPattern("MMM dd yyyy HHmm"));
+    }
+
+    public boolean isDone() {
```
Should this method have Javadocs?

georgetayqy reviewed (commented):
LGTM. There are a few nits and issues to iron out, but overall, good job!

Commit:
Add GUI functionality to the Yong chatbot

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI functionality to the Yong chatbot

Commit:
Edit files to follow checkstyle

Commit:
Merge branch 'A-CheckStyle'

* A-CheckStyle:
  Edit files to follow checkstyle

AaronJT1 commented on src/main/java/yong/Yong.java:
```diff
@@ -0,0 +1,73 @@
+     * @param userInput user String input
+     * @return Yong chatbot string response
+     */
+    public String getResponse(String userInput) {
```
I like how you use an intuitive variable name here 😄

AaronJT1 commented on src/main/java/yong/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,52 @@
+ */
+public class DeadlineCommand extends Command {
+
+    String inp;
```
Perhaps a more intuitive variable name here?

AaronJT1 commented on src/main/java/yong/tasks/Task.java:
```diff
@@ -0,0 +1,95 @@
+ */
+public class Task implements Serializable {
+    protected String description;
+    protected boolean isDone;
```
I like how you use a easy to recognise boolean name 😄

AaronJT1 commented on src/main/java/yong/command/EventCommand.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class EventCommand extends Command {
+
+    String inp;
```
Perhaps a more intuitive variable name here?

AaronJT1 reviewed (commented):
LGTM, Just a few nits to pick

Commit:
Add bye functionality to the GUI

Commit:
Update build gradle file to run GUI

Build gradle file is running the yong class instead of the launcher class for gui

Changed build gradle file to run the launcher class to launch the GUI

Whole project is being changed to be GUI based instead of CLI based

Commit:
Add assertions to the code

There are no assertions currently present in the code

Adding assertions to ensure that the code logic is correct at all steps and across modules

It is to ensure that there are no development bugs that are unforeseen

Commit:
Edit parser assertion function

Commit:
Revise code quality for different files

Code is not up to the coding quality standards in some methods

Extracted some methods and edited indentation to avoid deep nesting

To make the code more readable to the user

ngeeyonglim committed:
Merge pull request #1 from ngeeyonglim/branch-A-Assertions

Add assertions to the code

ngeeyonglim committed:
Merge pull request #2 from ngeeyonglim/branch-A-CodeQuality

Revise code quality for different files

Commit:
Add sort functionality to yong chatbot

Commit:
Merge branch 'branch-A-Sort'

* branch-A-Sort:
  Add sort functionality to yong chatbot

Commit:
Add java docs and fix nits from code review

Commit:
Add run method back in order to support Junit tests

Commit:
Edit javadocs

Commit:
Update Javadocs

Commit:
Edit readme and add UI.png

Commit:
Edit code to enable gradle

Commit:
Clean up Yong file

Commit:
Change name for Ui file

Commit:
Add new error handling for Event and Deadline dates

Commit:
Change JavaFx files to fix long list bug
